### DISCUSSION TITLE
Laravel setup for contribution, phpunit tests setup

### DISCUSSION THREAD

**Discussion by aprokopenko** [laravel/framework#47764] [Q&A] [answered]
Hi,

I can't find any documentation how to setup Laravel for Contributing locally. And the most important how to run framework tests!

I tried to download the framework repository, make `composer install, and then run `./vendor/bin/phpunit` and I get a lot of errors on clean Laravel install like 
```
1) Illuminate\Tests\Bus\BusBatchTest::test_jobs_can_be_added_to_the_batch
Illuminate\Database\QueryException: could not find driver 
```

If I install laravel/laravel - then tests just run app dummy folder, and not framework tests.

I want to submit a Pull Request, but I also want to run framework tests before I do that (because in case they fail, I will need to run them locally anyway)

**Comment by aprokopenko** [comment]
I got error in github tests, and now I really need to set it up locally, so would be great if someone can help((

**Comment by aprokopenko** [comment, marked as answer]
Update for those, who is also looking for this info:

1. Clone framework repo
2. Run `composer install`
3. Install docker / docker compose
4. Run `docker compose up -d` (there is `docker-compose.yml` with db and other supported engines)
5. Make sure you installed enough php packages, like mysql, pgsql, sqlite3, etc.! If some specific test failed - check the code for engine/technology it is used and install required php dependencies
6. Run the unit tests: `./vendor/bin/phpunit`

_For some reason I have memory limit error in the middle of the tests, when I run them all at once, maybe I didn't updated something correctly._

**Reply by DimaVIII** [reply]
Try to change memory_limit in your php.ini. Also check the memory setting for Docker.

**Reply by DimaVIII** [reply]
I just run the tests on macOS with:
```
git clone https://github.com/laravel/framework.git

cd framework 

composer install

./vendor/bin/phpunit
```

Result:
```
Time: 02:50.160, Memory: 323.50 MB

OK, but there were issues!
Tests: 8817, Assertions: 24407, Deprecations: 1, Skipped: 239.
```

It consumes 323.50 MB of Memory,  the default php.ini memory_limit is 128MB. You need to increase the limit.

On MacOS you can use `shivammathur/php` brew package which comes pre-set defaults in:

`cat /usr/local/etc/php/8.2/conf.d/php-memory-limits.ini`
```
; Max memory per instance
memory_limit = 512M

;The maximum size of an uploaded file.
upload_max_filesize = 512M

; Sets max size of post data allowed.
; Changes to this will also need to be reflected in Nginx with client_max_body_size
; This setting also affects file upload. To upload large files, this value must be larger than upload_max_filesize
post_max_size = 512M
```


### install
```shell
brew tap shivammathur/php
brew install shivammathur/php/php@8.2

# install xDebug
pecl install xdebug
```

**Comment by foertel** [comment]
Fails for me with broken dependencies on `10.x`. :/

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - laravel/prompts[v0.1.9, ..., v0.1.13] require illuminate/collections ^10.0|^11.0 -> found illuminate/collections[v10.0.0, ..., v10.37.3] but these were not loaded, likely because it conflicts with another require.
    - Root composer.json requires laravel/prompts ^0.1.9 -> satisfiable by laravel/prompts[v0.1.9, ..., v0.1.13].
```
